processor.go

```diff
@@ -609,7 +609,13 @@ func (g *Processor) Setup(session sarama.ConsumerGroupSession) error {
 	// create partition views for all partitions
 	for partition := range assignment {
 		// create partition processor for our partition
-		pproc, err := g.createPartitionProcessor(session.Context(), partition, runModeActive, session.MarkMessage)
+		pproc, err := g.createPartitionProcessor(session.Context(), partition, runModeActive,
+			func(msg *message, metadata string) {
+				// We have to commit the offset that we want to read next, not the one that we just have
+				// processed, therefore msg.offset+1 is required to get the next message.
+				// This has the same semantics as sarama's implementation of session.MarkMessage (which calls MarkOffset with offset+1)
+				session.MarkOffset(msg.topic, msg.partition, msg.offset+1, metadata)
```

Comment:
I still do not like the `+1` here in the processor. This is an offset-management detail and should be wrapped if possible.

Reply:
I tried to make it more clear in the next PR